services/brig/src/Brig/InternalEvent/Process.hs

```diff
@@ -23,6 +23,10 @@ onEvent n = handleTimeout $ case n of
         Log.info $ msg (val "Processing user delete event")
                 ~~ field "user" (toByteString uid)
         API.lookupAccount uid >>= mapM_ API.deleteAccount
+        -- As user deletions are expensive resource-wise in the context of
+        -- bulk user deletions (e.g. during team deletions),
+        -- wait 50ms before processing the next event
+        liftIO $ threadDelay 50000
```

Comment:
Make this configurable by a flag.

Shouldn't this be rate limited at a higher level than within an event handler?

Reply:
User deletions get pushed on an SQS queue. This event handler then pulls those events from a queue and processes them one by one. Queuing is used here mainly because user deletions entail a lot of deletions in different places. Without queuing, if while a user deletion is ongoing, a process crashes, for example, the user deletion would only be half-way done. This way, with events, upon a crash, the event will be retried (which is fine, deletion requests are idempotent in their end state). We want to ensure everything gets deleted if a user deletes their account.

What do you mean with "at a higher level" ?

Reply:
I think this can even be a little "slower", e.g., `100ms`.

Reply:
I was hoping whatever event library we use could have a per-endpoint rate limiting system. If that's not the case, then I guess this is okay for now.